File: frontend/src/app/core/query/reference-test.lookup.ts
import { Lookup } from "@common/model/lookup";
import { Guid } from "@common/types/guid";
import { Reference } from "../model/reference/reference";
import { ExternalFetcherBaseSourceConfigurationPersist } from "../model/external-fetcher/external-fetcher";

export class ReferenceTestLookup extends Lookup {
	like: string;
	key: string;
	dependencyReferences: Reference[];
	sources: ExternalFetcherBaseSourceConfigurationPersist[];

	constructor() {
		super();
	}
}

File: backend/core/src/main/java/org/opencdmp/query/lookup/ReferenceTestLookup.java
package org.opencdmp.query.lookup;

import gr.cite.tools.data.query.Lookup;
import org.opencdmp.model.persist.externalfetcher.ExternalFetcherBaseSourceConfigurationPersist;
import org.opencdmp.model.reference.Reference;

import java.util.List;

public class ReferenceTestLookup extends Lookup {

    private String like;

    private String key;

    private List<Reference> dependencyReferences;

    private List<ExternalFetcherBaseSourceConfigurationPersist> sources;

    public String getLike() {
        return like;
    }

    public void setLike(String like) {
        this.like = like;
    }

    public String getKey() {
        return key;
    }

    public void setKey(String key) {
        this.key = key;
    }

    public List<Reference> getDependencyReferences() {
        return dependencyReferences;
    }

    public void setDependencyReferences(List<Reference> dependencyReferences) {
        this.dependencyReferences = dependencyReferences;
    }

    public List<ExternalFetcherBaseSourceConfigurationPersist> getSources() {
        return sources;
    }

    public void setSources(List<ExternalFetcherBaseSourceConfigurationPersist> sources) {
        this.sources = sources;
    }
}
